classes/AlertView.php

```diff
@@ -123,6 +129,26 @@ private function confirmAlert($token) {
         return $this->alert->confirm($token);
     }
 
+    private function replaceAlert($confirmation) {
+        $existing = $this->alert->fetch_by_token($confirmation);
```

Comment:
The alert script will, if I've understood it right, have found any alert containing `speaker:ID` (e.g. "horse speaker:ID"), not just exactly `speaker:ID`. As well as dealing with that, this script needs to replace all such alerts the user has, not just that particular one whose token it put in the email.